relay-sampling/src/lib.rs

```diff
@@ -441,23 +441,25 @@ impl FieldValueProvider for Event {
                 }
                 _ => Value::Null,
             },
-            "transaction.op" => match (self.ty.value(), store::get_transaction_op(self)) {
+            "event.contexts.trace.op" => match (self.ty.value(), store::get_transaction_op(self)) {
                 (Some(&EventType::Transaction), Some(op_name)) => Value::String(op_name.to_owned()),
                 _ => Value::Null,
             },
             "event.platform" => match self.platform.value() {
                 Some(platform) if store::is_valid_platform(platform) => {
                     Value::String(platform.to_owned())
                 }
-                _ => Value::Null,
+                _ => Value::String("other".to_owned()),
             },
-            field_name if field_name.starts_with("transaction.measurements.") => {
-                let measurement_name = &field_name["transaction.measurements.".len()..];
-                if let Some(value) = store::get_measurement(self, measurement_name) {
-                    value.into()
-                } else {
-                    Value::Null
-                }
+            field_name
+                if field_name.starts_with("event.measurements.")
+                    && field_name.ends_with(".value") =>
+            {
+                field_name
+                    .get("event.measurements.".len()..field_name.len() - ".value".len())
```

Comment:
nit: I wonder if at this point matching a regex group would be easier to read. Would theoretically even be more efficient because you'd only have to iterate the string once.

Reply:
i was thinking the same. but fwiw we're not iterating here: `"foo".len()` is optimized to an integer, and the slicing operation is just pointer math

Reply:
Oh you're right.